### PR TITLE
Add content-unclosed-fence: detect unterminated code fences that blind content rules

### PR DESCRIPTION
**Pull request by not-stbenjam**
## What

A new content rule, `content-unclosed-fence` (warning, `enabled: auto`, since 0.17.0), that detects a code fence opened with ``` ``` ``` or `~~~` but never closed, plus a SUGGEST-confidence autofix that appends the matching closing fence at end of file.

Supporting change: `MarkdownDoc.fences()` now populates two new `MarkdownFence` fields — `markup` (the opening fence run, e.g. ``` ``` ``` or `~~~~`) and `nested` (inside a blockquote/list container) — which the closedness check needs.

## Why

An unterminated fence makes markdown-it treat **everything after it as code**. `read_body(strip_code_blocks=True)` then strips it, so every content rule is blinded: a SKILL.md whose unclosed fence swallows weak-language prose lints clean and grades A+ with zero violations. The included fixture demonstrates exactly this — before the fence is closed, `content-weak-language` reports nothing; close the fence where the code actually ends and three hedging violations surface.

The same defect also breaks rendering for humans and agents alike (GitHub shows the swallowed sections as one giant code block), so this is a real authoring bug, not just a linter blind spot.

## How

- **Detection** (`src/skillsaw/rules/builtin/content/unclosed_fence.py`): only the last fence of a body can reach EOF, and an unclosed top-level fence always does. Flag it when its last line is not a valid CommonMark closer (run of >= opener length of the same fence character, <= 3 spaces indent, only whitespace after). Reports the opening fence line. Distinguishes cleanly: tilde fences, longer closer runs, ```` ```` ````-outer/``` ``` ```-inner, closers with trailing spaces/indent, pseudo-closers with info strings, fences closed exactly at EOF (with or without trailing newline), indented code blocks.
- **Scope**: container-nested fences (blockquote/list) are skipped — a column-0 closer would open a *new* fence instead of terminating them, so neither detection nor the fix can act safely there.
- **Autofix (SUGGEST)**: appends the matching closer (same char, same length) as a new final line — the only splice guaranteed not to change how the file already renders. The fix description and docs tell the reviewer to move the closer up if the code block was meant to end earlier; that review step is what resurfaces the hidden prose. Skips YAML-embedded bodies (`.coderabbit.yaml` instructions, promptfoo prompts) where a bare closer at EOF would corrupt the host document. Idempotent and convergent.

One deliberate deviation from a naive reading of "the fix surfaces the hidden violations": an EOF-appended closer necessarily leaves the swallowed prose *inside* the now-closed fence, so hidden violations only surface after the human review step moves the closer to the intended line. The integration tests encode both halves honestly (fix converges + closing at the intended line surfaces 3 weak-language violations).

## Testing

- Unit: `TestFences` additions for `markup`/`nested` (`tests/test_markdown_doc.py`); `TestContentUnclosedFenceRule` (17 detection cases incl. all edge cases above) and `TestContentUnclosedFenceAutofix` (`tests/test_content_rules.py`).
- Integration (`tests/test_integration.py`, fixture `tests/fixtures/content-unclosed-fence/skill-hides-violations/`): detection at the right line, the blindness regression (weak language absent before / present after the fence closes where the code ends), plain `fix` does not apply SUGGEST fixes, `fix --suggest` appends exactly one closing line and converges, idempotency. Fixture added to `BROKEN_FIXTURES` for rule-coverage enforcement and carries a `skillsaw-assert` directive.
- `make test`: 2504 passed. `make lint` clean. `make update` regenerated `.skillsaw.yaml.example` + site docs (included).
- False-positive sweep on `openshift-eng/ai-helpers` (40 plugins, 103 skills): with their pinned config, exit 0 (rule version-gated by `since`). Force-enabled, the rule finds exactly 3 hits — all manually verified true positives with genuinely broken fence structure (a heredoc at column 0 breaking a list-nested fence; an outer fence "closed" only by info-string openers like ```` ```text ````; a desynced closer swallowing a `## See Also` section). Zero false positives there and on skillsaw itself.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added support for detecting Markdown code fences that are opened but never closed.
  * Added guidance and examples for fixing unclosed fences, along with a new rule entry in the rules catalog.
  * Included autofix support that can add the missing closing fence in eligible cases.

* **Bug Fixes**
  * Improved parsing of fenced code blocks, including different fence styles and nested contexts.
  * Added end-to-end coverage so hidden content is surfaced again after the fence is corrected.

### CODE REVIEW

File: src/skillsaw/rules/builtin/content/unclosed_fence.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+                continue
+            content = read_text(cf.path)
+            if content is None:
+                continue
+            # Append only when the body ends the file (plain markdown files
+            # and frontmattered bodies).  YAML-embedded bodies (.coderabbit
+            # instructions, promptfoo prompts) are indented inside a host
+            # document — a bare closer at EOF would corrupt it.
+            if not content.endswith(body):
+                continue
```
[the remark this answers is not in the document]
Comment (not-stbenjam) [reply src/skillsaw/rules/builtin/content/unclosed_fence.py]: Addressed in 2f531ed.

File: src/skillsaw/rules/builtin/content/unclosed_fence.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        """True when *line* is a valid CommonMark closer for a fence opened
+        with *markup*: at most 3 spaces of indent, then a run of the same
+        fence character at least as long as the opener, then only whitespace.
+        """
+        text = line.rstrip()
+        run = text.lstrip(" ")
+        if len(text) - len(run) > 3:
+            return False
+        return len(run) >= len(markup) and run == markup[0] * len(run)
+
```
[the remark this answers is not in the document]
Comment (not-stbenjam) [reply src/skillsaw/rules/builtin/content/unclosed_fence.py:97]: Addressed in 2f531ed.

**Comment by stbenjam** [comment]
I am skeptical of the value of this one over just a mark down linter

**Comment by stbenjam** [comment]
Looking at the rule impact it is finding legit problems, this one rule probably makes sense for us to have.
